### PR TITLE
fix: adds an action when only one of the required for MCT was ignored

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
when you have a remote-only Apex class like Class1
and have the entry `**/Class1.cls` in the `.forceignore` file, a retrieve would error with 
`ERROR running force:source:retrieve: Metadata API request failed: ApexClass types require source to be present and unpackaged/classes/Class1.cls is forceignored.`

new message:
`Metadata API request failed: ApexClass types require source to be present and unpackaged/classes/Class1.cls is forceignored. 
 ACTION: Both the content file, and -meta.xml file must be ignored, try appending * to your .forceignore entry`

this could lead to a confusing behavior if you're new to sfdx dev. "I ignored the class, why can't I retrieve it", it's because that metadata type _actually_ has two files that both must be ignored

this adds a message to ignore the `.cls`(example) and `-meta.xml` file for all MatchingContentType metadata types

### What issues does this PR fix or reference?

@W-10037182@ https://github.com/forcedotcom/cli/issues/1238

### CODE REVIEW

File: src/i18n/i18n.ts [hunk 1]
```diff
@@ -43,7 +43,8 @@ export const messages = {
   error_missing_type_definition: "Missing metadata type definition in registry for id '%s'",
   error_missing_child_type_definition: 'Type %s does not have a child type definition %s',
   error_no_metadata_xml_ignore: 'Metadata xml file %s is forceignored but is required for %s',
-  error_no_source_ignore: '%s types require source to be present and %s is forceignored.',
+  error_no_source_ignore:
+    '%s types require source to be present and %s is forceignored. \n ACTION: Both the content file, and -meta.xml file must be ignored, try appending * to your .forceignore entry',
```
Comment (mshanemc) [inline src/i18n/i18n.ts:47]:
```suggestion
    '%s types require source to be present and %s is forceignored. \n ACTION: Both the content file and -meta.xml file must be ignored.  Try appending * to your .forceignore entry',
```

**Review by mshanemc — APPROVED** [review]

**Review by randi274 — APPROVED** [review]

File: src/i18n/i18n.ts [hunk 2]
```diff
@@ -43,7 +43,8 @@ export const messages = {
   error_missing_type_definition: "Missing metadata type definition in registry for id '%s'",
```
Comment (jshackell-sfdc) [inline src/i18n/i18n.ts:43]:
```suggestion
  error_missing_type_definition: "Missing metadata type definition in registry for id '%s'.",
```

File: src/i18n/i18n.ts [hunk 3]
```diff
@@ -43,7 +43,8 @@ export const messages = {
...
   error_missing_child_type_definition: 'Type %s does not have a child type definition %s',
```
Comment (jshackell-sfdc) [inline src/i18n/i18n.ts:44]:
```suggestion
  error_missing_child_type_definition: 'Type %s does not have a child type definition %s.',
```

File: src/i18n/i18n.ts [hunk 4]
```diff
@@ -43,7 +43,8 @@ export const messages = {
...
   error_no_metadata_xml_ignore: 'Metadata xml file %s is forceignored but is required for %s',
```
Comment (jshackell-sfdc) [inline src/i18n/i18n.ts:45]:
```suggestion
  error_no_metadata_xml_ignore: 'Metadata xml file %s is forceignored but is required for %s.',
```

File: src/i18n/i18n.ts [hunk 5]
```diff
@@ -56,7 +57,7 @@ export const messages = {
   error_no_directory_stream: '%s does not support readable streams on directories',
```
Comment (jshackell-sfdc) [inline src/i18n/i18n.ts:57]:
```suggestion
  error_no_directory_stream: '%s doesn\'t support readable streams on directories.',
```

File: src/i18n/i18n.ts [hunk 6]
```diff
@@ -56,7 +57,7 @@ export const messages = {
...
   error_no_source_to_deploy: 'No source backed components present in the package',
```
Comment (jshackell-sfdc) [inline src/i18n/i18n.ts:58]:
```suggestion
  error_no_source_to_deploy: 'No source-backed components present in the package.',
```

File: src/i18n/i18n.ts [hunk 7]
```diff
@@ -56,7 +57,7 @@ export const messages = {
...
   error_no_components_to_retrieve: 'No components in the package to retrieve',
```
Comment (jshackell-sfdc) [inline src/i18n/i18n.ts:59]:
```suggestion
  error_no_components_to_retrieve: 'No components in the package to retrieve.',
```

File: src/i18n/i18n.ts [hunk 8]
```diff
@@ -56,7 +57,7 @@ export const messages = {
...
-  error_invalid_package: 'The metadata pacakge was not initialized properly',
+  error_invalid_package: 'The metadata package was not initialized properly',
```
Comment (jshackell-sfdc) [inline src/i18n/i18n.ts:60]:
```suggestion
  error_invalid_package: 'The metadata package wasn\'t properly initialized.',
```

File: src/i18n/i18n.ts [hunk 9]
```diff
@@ -56,7 +57,7 @@ export const messages = {
...
   error_static_resource_expected_archive_type:
     'A StaticResource directory must have a content type of application/zip or application/jar - found %s for %s',
```
Comment (jshackell-sfdc) [inline src/i18n/i18n.ts:62]:
```suggestion
    'A StaticResource directory must have a content type of application/zip or application/jar - found %s for %s.',
```

**Review by jshackell-sfdc — CHANGES_REQUESTED** [review]

**Review by cristiand391 — APPROVED** [review]
QA:
Created and empty project, created apex class `Class1` and `Class2` from org ui, forceignore file:
```
...
**/Class1.cls
```
```
sfdx force:source:retrieve --metadata=ApexClass
ERROR running force:source:retrieve:  Metadata API request failed: ApexClass metadata types require source files, but unpackaged/classes/Class1.cls is forceignored.
        ACTION: Metadata types with content are composed of two files: a content file (ie MyApexClass.cls) and a -meta.xml file (i.e MyApexClass.cls-meta.xml). You must include both files in your .forceignore file. Or try appending “*” to your existing .forceignore entry.
```

**Pull request merged**